src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -372,7 +376,7 @@ protected function createNode()
                 $node->setKeyAttribute($this->key, $this->removeKeyItem);
             }
 
-            if (true === $this->atLeastOne) {
+            if (true === $this->atLeastOne || false === $this->allowEmptyValue) {
```

Comment:
Updated.

Removed the deprecation (worth it?) so both `requiresAtLeastOneElement` and `cannotBeEmpty` act the same. We cannot really deprecate/remove cannotBeEmpty as it comes from the base class (not sure this should be reversed in any way from a sub class).

So either deprecate `requiresAtLeastOneElement` in favor of `cannotBeEmpty` or support both like i did.